aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -411,11 +416,65 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   return at::_index_copy_(self, dim, index, source);
 }
 
+Tensor & _index_copy_impl_(Tensor & self, int64_t dim, const Tensor & index, const Tensor & source) {
+  // Handle the case when `self` is 0-dim
+  Tensor self_nonzero = self.dim() == 0 ? self.unsqueeze(-1) : self;
```

Comment:
@ngimel: this seems odd. What's going on here? Won't this result in an out-of-place operation since self_nonzero will be a different tensor? More precisely, while the data might be manipulated in place, self_nonzero would be a distinct tensor object from self.

Reply:
nevermind, `self` is ultimately returned, so all's good.